In [1]:
import pandas
from pandas import DataFrame 
import numpy

In [2]:
dataset: DataFrame = pandas.read_csv("data/caso_full.csv")

## General nan input

In [3]:
DataFrame(dataset.isna().sum().sort_values(ascending = False))

,0
last_available_confirmed_per_100k_inhabitants,29166
city,20119
estimated_population,13646
estimated_population_2019,13646
city_ibge_code,13646
date,0
epidemiological_week,0
is_last,0
is_repeated,0
last_available_confirmed,0


In [4]:
dataset = dataset.replace(
    ["<NA>", "NA", "NaN", "nan", "null", "", "Importados/Indefinidos"],
    numpy.nan
)

In [5]:
DataFrame(dataset.isna().sum().sort_values(ascending = False))

,0
city,33765
last_available_confirmed_per_100k_inhabitants,29166
estimated_population,13646
estimated_population_2019,13646
city_ibge_code,13646
date,0
epidemiological_week,0
is_last,0
is_repeated,0
last_available_confirmed,0


### Missing columns rate

In [ ]:
missing_columns = [
    "city",
    "last_available_confirmed_per_100k_inhabitants",
    "estimated_population",
    "estimated_population_2019",
    "city_ibge_code"
]

missing_summary = DataFrame({
    "Missing Count": dataset[missing_columns].isna().sum(),
    "Missing Rate": dataset[missing_columns].isna().mean()
}).sort_values("Missing Rate", ascending = False)

display(
    missing_summary
    .style
    .format({"Missing Rate": "{:.2%}"})
    .background_gradient(cmap = "Reds", subset = ["Missing Rate"])
    .set_caption("Missing Values Summary")
)

,Missing Count,Missing Rate
city,33765,0.88%
last_available_confirmed_per_100k_inhabitants,29166,0.76%
estimated_population,13646,0.35%
estimated_population_2019,13646,0.35%
city_ibge_code,13646,0.35%


In [10]:
rows_missing = dataset[missing_columns].isna().any(axis = 1)

summary_stats = DataFrame({
    "Metric": ["Rows with missing values", "Rate of affected rows"],
    "Value": [
        f"{rows_missing.sum():,}",
        f"{rows_missing.mean():.2%}"
    ]
})

display(
    summary_stats
    .style
    .set_caption("Dataset Impact Summary")
)

,Metric,Value
0,Rows with missing values,"49,279"
1,Rate of affected rows,1.28%


## Since rows with missing values represent only 1.28% of the dataset, they are removed.